crypto-bigint/src/traits.rs

```diff
@@ -1,6 +1,81 @@
 //! Traits provided by this crate
 
 use crate::Limb;
+use core::fmt::Debug;
+use subtle::{
+    Choice, ConditionallySelectable, ConstantTimeEq, ConstantTimeGreater, ConstantTimeLess,
+};
+
+#[cfg(feature = "rand")]
+use rand_core::{CryptoRng, RngCore};
+
+/// Integer type.
+pub trait Integer:
```

Comment:
why no requirements for Add, Mul etc?

Reply:
Those traits are presently only impl'd for `Wrapping<UInt<_>>` in order to be explicit about the wrapping vs checked behavior.

Reply:
FWIW, in `elliptic-curve` I end up bounding on the modular arithmetic traits: https://github.com/RustCrypto/traits/pull/732/files#diff-58bf09549ae77b47fcc1f18c3335fd0f6a4eb49a1e3cee51332fe40dcfd66daaR113-R120

```rust
type UInt: bigint::AddMod<Output = Self::UInt>
        + bigint::Integer
        + bigint::NegMod<Output = Self::UInt>
        + bigint::Random
        + bigint::RandomMod
        + bigint::SubMod<Output = Self::UInt>
        + zeroize::Zeroize;
```

They could potentially be included in the `Integer` bounds.

Reply:
I guess I didn't add bound on `*Mod` initially because they are only impl'd for a subset of the supported integer values

Reply:
Yeah makes sens to wait for adding those bound, until we have a more complete set